### DISCUSSION TITLE
How does a new Dagster deployment initialize the schema and tables of its database? Is it automatic or manual?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18207] [Q&A] [answered]
i have dagster deployed via docker. the postgres database the application needs for run, schedule and event_log storage is hosted in a separate docker container (pretty much the exact setup in the dagster docs). i'd like to move this database into an RDS solution hosted on AWS. once i have the new database up-and-running, will i need to actually create the schema and tables for the application to write into? or will it create them automatically once i update the instance's database connection? i don't care about preserving history from the docker container.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700603099081989?thread_ts=1700603099.081989&cid=C01U954MEER).*

**Comment by gibsondan** [comment, marked as answer]
Dagster will automatically initialize the schema and tables in RDS the first time either the daemon or the webserver start up.

**Reply by kaush-13** [reply]
do you know if there is a way to specify the schema if you want to use something other than public?

**Reply by vergenzt** [reply]
FYI @kaush-13 as far as I know there's not currently a way to specify a different schema. I've made a feature request here though which you can upvote! https://github.com/dagster-io/dagster/issues/25312
